### PR TITLE
Fix users import during installation

### PR DESCRIPTION
**Pull request by imobach**
Fixes bsc#973639 and bsc#971804.

### CODE REVIEW

File: src/clients/users_finish.rb [hunk 1]
```diff
@@ -71,9 +73,12 @@ def main
 
           # During installation, some package could add a new user, so we
           # need to read them again before writing.
```
Comment (kobliha) [inline src/clients/users_finish.rb]: Maybe the comment does not belong here, but down ... above the `Read()`?

File: src/clients/users_finish.rb [hunk 2]
```diff
@@ -71,9 +73,12 @@ def main
...
-          Users.SetExportAll(true)
+          Users.SetExportAll(false) # Export only previously imported users
```
Comment (kobliha) [inline src/clients/users_finish.rb]: And then you could move the comment just above?

File: src/clients/users_finish.rb (hunk 1 above)
Comment (imobach) [inline src/clients/users_finish.rb]: What I wanted to express is that we need to Export -> Re-read -> Import again (that's the reason to put the comment before the whole block).

But I see that's not clear at all, so I've added some extra documentation.

**Comment by schubi2** [comment]
Uhi, you have fixed a lot of bnc#* !  :-) LGTM

File: src/clients/users_finish.rb [hunk 3]
```diff
@@ -71,12 +73,20 @@ def main
...
+          Users.SetExportAll(false)
           saved = Users.Export
-          Users.ReadLocal
+          log.info("Users to import: #{saved}")
+
+          # 2. Read users and settings from the installed system
+          # (bsc#965852, bsc#973639, bsc#974220 and bsc#971804)
+          Users.Read
+
+          # 3. Import users exported in step 1
```
Comment (kobliha) [inline src/clients/users_finish.rb]: It's actually `Merge users from the system with new users from AutoYast profile` ?

**Comment by imobach** [comment]
I've moved `UsersFinish` client to `lib/users/clients`. BTW, I've done some refactoring and now it inherits from `::Installation::FinishClient`.

Adding tests was a little bit tricky because the code, as it's right now, is not easy to test. But I hope it's better now :)

**Comment by kobliha** [comment]
LGTM (but I haven't checked whether it all works "as is", Travis should say his word here)

**Comment by mvidner** [comment]
The matching integration test is https://github.com/yast/aytests-tests/pull/19.

File: package/yast2-users.spec [hunk 4]
```diff
@@ -37,6 +37,7 @@ BuildRequires:  yast2-ldap >= 3.1.2
 BuildRequires:  yast2-perl-bindings
 BuildRequires:  yast2-security
 BuildRequires:  yast2-testsuite
+BuildRequires:  rubygem(rspec)
```
Comment (mvidner) [inline package/yast2-users.spec]: `BuildRequires:  rubygem(%rb_default_ruby_abi:rspec)`

**Comment by mvidner** [comment]
Thanks for the client refactoring and tests!
Just one packaging improvement,
otherwise LGTM.

**Comment by imobach** [comment]
@mvidner Thanks! I've updated the spec file with your suggestion.

**Comment by mvidner** [comment]
LGTM

**Comment by imobach** [comment]
Thanks both for reviews. Merging after getting positive feedback.

**Pull request merged**